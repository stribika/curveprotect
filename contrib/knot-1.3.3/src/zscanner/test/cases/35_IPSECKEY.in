$ORIGIN	.
$TTL	1

; OK
@	IPSECKEY	0	0	0	.			; The simplest variant - no gw, no key
@	IPSECKEY	255	3	255	.	AA==		; Maximal numbers
@	IPSECKEY	0	1	0	0.0.0.0			; IPv4 address
@	IPSECKEY	0	2	0	::			; IPv6 address
@	IPSECKEY	0	3	0	\0320\ \\\"\.\@\*.tld.	; Special chars in domain name
@	IPSECKEY	0	0	1	.	Zm8=            ; One char padding
@	IPSECKEY	0	0	1	.	Zm9v            ; Without padding
@	IPSECKEY	0	0	1	.	Zm9vYg==        ; Two base64 blocks
@	IPSECKEY	0	0	1	.	Zm9v YmE=       ; Two base64 blocks with blank space between them
@	TYPE45	\# 3 000000						; TYPE + Hexadecimal rdata
@	TYPE45		0	0	1	.	AA==		; TYPE
@	ipseckey	0	0	1	.	AA==		; Type in lower-case

; KO
@	IPSECKEY
@	IPSECKEY							; Empty rdata
@	IPSECKEY	\# 0						; Hex empty rdata
@	IPSECKEY	\#						; Missing hex length
@	IPSECKEY	256	0	0	.			; Precedence overflow
@	IPSECKEY	0	4	0	.			; Unknown gateway
@	IPSECKEY	0	0	256	.	AA==		; Algorithm overflow
@	IPSECKEY	0	0	0	.	AA==		; If alg is 0 then key shouldn't be given
@	IPSECKEY	0	0	0	a%			; Bad domain name char
@	IPSECKEY	0	0	1	.	A		; Continuous block length must be multiple of 4
@	IPSECKEY	0	0	1	.	=		; Bad padding
@	IPSECKEY	0	0					; Missing item
@	IPSECKEY	\# 3 00000000					; Too long rdata
@	IPSECKEY	\# 4 000000					; Bad rdata length
