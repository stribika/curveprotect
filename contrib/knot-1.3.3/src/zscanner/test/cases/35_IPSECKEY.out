OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=FF03FF0000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=00010000000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=00020000000000000000000000000000000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000300082030205C222E402A03746C6400
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000001666F
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000001666F6F
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000001666F6F62
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000001666F6F6261
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=00000100
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002D
RDATA=00000100
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_HEX_CHAR
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_ENUMBER8_OVERFLOW
------
WARNG=ZSCANNER_EBAD_GATEWAY
------
WARNG=ZSCANNER_ENUMBER8_OVERFLOW
------
WARNG=ZSCANNER_EBAD_GATEWAY_KEY
------
WARNG=ZSCANNER_EBAD_GATEWAY
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_GATEWAY_KEY
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
